src/test/java/com/epam/tc/hw1/DivTest.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.tc.hw1;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class DivTest extends AbstractTest {
+
+    @DataProvider
+    public static Object[][] divCorrectData() {
+        return new Object[][]{
+                {10.0, 2.0, 5.0},
+                {17.0, 5.0, 3.4},
+                {4, 5, 0.8},
+                {0, 10, 0},
+                {25, 5, 5}
+        };
+    }
+
+    @Test(dataProvider = "divCorrectData")
+    public void positiveDivTest(double a, double b, double result) {
+        var act = calculator.div(a, b);
+        assertThat(act).isEqualTo(result);
+    }
+
+    @Test(expectedExceptions = NumberFormatException.class)
+    public void negativeDivOnZeroLongTest() {
+        long a = 922337203685772107L;
+        long b = 0;
+        long result = 0;
+        var act = calculator.div(a, b);
+        assertThatThrownBy(() -> calculator.div(a, b)).isInstanceOf(NumberFormatException.class);
+    }
+
+    @Test
+    public void negativeDivOnZeroDoubleTest() {
+        double a = 9223.0;
+        double b = 0.0;
+        double result = 0.0;
+        var act = calculator.div(a, b);
+        softAssert.assertEquals(act, result, 0.0001);
+    }
```

Comment:
1. double(0.0 / doube(0.0) = double(Infinity) != 9223.0
2. softAssert is useless without `assertAll()` method. 
3. Why are you using softAssert here? Why not hard assert?

Reply:
Sorry, double(0.0 / doube(0.0) = double(NAN) != 9223.0
double(1.0 / doube(0.0) = double(Infinity) != 9223.0